### ISSUE TITLE
Prototype base

### ISSUE THREAD

**Issue by nickatnight** [WallStreetAnalytics/edgar-parsing-prototype#1]
See #devops in discord for specs

**Comment by yololftw** [comment]
Its pretty hard to follow spec on Discord.

Let me port it here. 


If anyone is interested in whipping out a prototype for extracting data from EDGAR, here's the details (If no one gets to it before I do, I'll knock it out sometime in the next few days but wanted to give others a chance if they are interested and want to):

1. We'll want it in Python since there are libraries already available to do a lot of the heavy lifting (https://pypi.org/project/python-edgar/)
2. That lib will download the index zip files that are | delimited. We loop through those and then do a GET for the URLs that correspond with 10-K and 10-Qs
3. Parse the txt docs that come back (sample: https://www.sec.gov/Archives/edgar/data/1001039/0001001039-17-000141.txt) to extract balance sheet, income statements, and cash flow statements. Here are some resources of others doing similar things: https://github.com/lukerosiak/pysec https://gist.github.com/anshoomehra/ead8925ea291e233a5aa2dcaa2dc61b2
4. Store that data in a database. The easiest thing for this rapid prototype would be a mongodb collection, so you don't have to worry about structure so much. 
5. Use Docker-Compose to make it easy and fast for others to spin up the prototype and find data.

**Resources**
[Parse XBRL filings from the SEC's EDGAR in Python](https://gist.github.com/anshoomehra/ead8925ea291e233a5aa2dcaa2dc61b2)
[ Parse XBRL filings from the SEC's EDGAR in Python ](https://github.com/lukerosiak/pysec)

**Comment by kginstructor** [comment]
Did anyone get a chance to test @nickatnight 's prototype yet?
